14_structures/exercises/booksave_delete.c

```diff
@@ -0,0 +1,121 @@
+// Modify Listing 14.14 so that as each record is read from the file and shown to you, you
+// are given the chance to delete the record or to modify its contents. If you delete the
+// record, use the vacated array position for the next record to be read. To allow changing
+// the existing contents, you'll need to use the "r+b" mode instead of the "a+b" mode, and
+// you'll have to pay more attention to positioning the file pointer so that appended records
+// don't overwrite existing records. It's simplest to make all changes in the data stored in
+// program memory and then write the final set of information to the file. One approach to
+// keeping track is to add a member to the book structure that indicates whether it is to be
+// deleted.
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define MAXTITL 40
+#define MAXAUTL 40
+#define MAXBKS 10
+
+char * s_gets(char * st, int n);
+
+struct book {
+    char title[MAXTITL];
+    char author[MAXAUTL];
+    float value;
+};
+
+int main (void)
+{
+    struct book library[MAXBKS];
+    int count = 0;
+    int option = 0;
+    int index;
+    FILE * pbooks;
+    int size = sizeof(struct book);
+
+    pbooks = fopen("book.dat", "r+b");
```

Comment:
When file does not exists the program breaks. Instead it should create the file and ask to add books.